src/marqo/tensor_search/tensor_search.py

```diff
@@ -274,17 +274,38 @@ def get_stats(config: Config, index_name: str):
 
     Difference between the two gives the numberOfVectors."""
 
+    body = {
+        "size": 0,
+        "aggs": {
+            "nested_chunks": {
+                "nested": {
+                    "path": "__chunks"
+                },
+                "aggs": {
+                    "chunk_count": {
+                        "value_count": {
+                            "script": {
+                                "source": "params._source['__chunks'] != null ? params._source['__chunks'].length : 0"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
     try:
         doc_count = HttpRequests(config).post(path=F"{index_name}/_count")["count"]
-        nested_doc_count = HttpRequests(config).get(path=F"{index_name}/_stats/docs")["indices"][index_name]["primaries"][
-            "docs"]["count"]
+        vector_count = HttpRequests(config).get(path=f"{index_name}/_search", body=body) \
+            ["aggregations"]["nested_chunks"]["chunk_count"]["value"]
     except KeyError as e:
```

Comment:
nice job catching KeyError.

You may also need to catch TypeError, if you are parsing an unknown dict